### PR TITLE
SAK-29694 Filtered Requested Sites

### PR DESCRIPTION
**Pull request by puramshetty**

### CODE REVIEW

**Comment by seththeriault** [comment]
I have added the Jira reference to the title to conform to general practice.

**Comment by buckett** [comment]
Thanks for the patch. For doing an intersection in Java you can use `Collection.retainAll(Collection)`.

File: polls/impl/src/java/org/sakaiproject/poll/service/impl/PollListManagerImpl.java [hunk 1]
```diff
@@ -126,28 +127,16 @@ public void destroy() {
...
-                        if (!found) {
-                            allowedSites.remove(j);
-                        }
-                    }
-                } else {
-                    // no sites to search so EXIT here
-                    return new ArrayList<Poll>();
-                }
-            }
-            String[] siteIdsToSearch = allowedSites.toArray(new String[allowedSites.size()]);
```
Comment (steveswinsburg) [inline polls/impl/src/java/org/sakaiproject/poll/service/impl/PollListManagerImpl.java]: This is a strange and complex iteration, why not just:
`
List<String> requestedSiteIds = Arrays.asList(siteIds);
allowedSites.retainAll(requestedSiteIds);
`

File: portal/portal-render-engine-impl/pack/src/webapp/vm/morpheus/includeAnalytics.vm [hunk 2]
```diff
@@ -34,8 +34,7 @@
         m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)
         })(window,document,'script','//www.google-analytics.com/analytics.js','ga');
 
-        ga('create', '${googleUniversalAnalyticsId}', 'auto');
-        ga('create', '${googleUniversalAnalyticsId}', { 'userId': portal.user.eid });
+        ga('create', '${googleUniversalAnalyticsId}', { 'userId': portal.user.id });
```
Comment (steveswinsburg) [inline portal/portal-render-engine-impl/pack/src/webapp/vm/morpheus/includeAnalytics.vm]: unrelated modification. Please revert.

**Comment by puramshetty** [comment]
@steveswinsburg I am new to github and struggling. Can you please confirm whether the reverts are fine? Please help me out.

**Comment by steveswinsburg** [comment]
Yeah that line that was added is now gone, but after this PR you might want to read up on git and how it treats the commits.

Last outstanding bit is to fix up the iteration code and then we can get this in. Thanks for your hard work on this!

**Comment by buckett** [comment]
I've taken your commit and added a test for it on a new pull request #1253 and cleanup up the indentation.
You're still the author on the fix.

**Comment by buckett** [comment]
Thanks for digging into this and finding the problem.
